cmake/TileDB-Superbuild.cmake

```diff
@@ -128,16 +128,31 @@ endif()
 # Set up the regular build (i.e. non-superbuild).
 ############################################################
 
-ExternalProject_Add(tiledb
-  SOURCE_DIR ${PROJECT_SOURCE_DIR}
-  CMAKE_ARGS
-    -DTILEDB_SUPERBUILD=OFF
-    ${INHERITED_CMAKE_ARGS}
-    ${FORWARD_EP_CMAKE_ARGS}
-  INSTALL_COMMAND ""
-  BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR}/tiledb
-  DEPENDS ${TILEDB_EXTERNAL_PROJECTS}
-)
+if(WIN32)
+  ExternalProject_Add(tiledb
+    SOURCE_DIR ${PROJECT_SOURCE_DIR}
+    CMAKE_ARGS
+      -DTILEDB_SUPERBUILD=OFF
+      ${INHERITED_CMAKE_ARGS}
+      ${CMAKE_CXX_FLAGS}
+      -DCURL_STATICLIB=ON
```

Comment:
Is this necessary for the build? We have a number of different distribution targets, and libcurl may be a shared library in some cases (eg: conda). Ideally the definition should be set on the `CURL::libcurl` target, although that might not always be available for a non-cmake curl install. cc @Shelnutt2, I think at minimum this should only be set when `TILEDB_STATIC` is true.

Reply:
Not currently need to build, removed.